### PR TITLE
Implemented basic Devise account creation

### PR DESCRIPTION
**Pull request by apfong**
Adds a sign up and login button in the navbar. Currently the accounts are being
stored in a sqlite3 database, but it should work even after moving to mysql.

### CODE REVIEW

File: config/initializers/devise.rb [hunk 1]
```diff
@@ -0,0 +1,262 @@
+# Use this hook to configure devise mailer, warden hooks and so forth.
+# Many of these configuration options can be set straight in your model.
+Devise.setup do |config|
+  # The secret key used by Devise. Devise uses this key to generate
+  # random tokens. Changing this key will render invalid all existing
+  # confirmation, reset password and unlock tokens in the database.
+  # Devise will use the `secret_key_base` on Rails 4+ applications as its `secret_key`
+  # by default. You can change it below and use your own secret key.
+  # config.secret_key = '7e029e4ded76273ff6d2f36dd6c9694c61d646ec1301bd934cc7312c1697099a6762bbb60af45278cf02ee083bd0ddb6ff3ff5f60972cf92dfe3cb0380b00cd9'
```
Comment (bmwwoo) [inline config/initializers/devise.rb]: Put this in an ENV var

File: config/initializers/devise.rb [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+  #
+  # Limiting the stretches to just one in testing will increase the performance of
+  # your test suite dramatically. However, it is STRONGLY RECOMMENDED to not use
+  # a value less than 10 in other environments. Note that, for bcrypt (the default
+  # encryptor), the cost increases exponentially with the number of stretches (e.g.
+  # a value of 20 is already extremely slow: approx. 60 seconds for 1 calculation).
+  config.stretches = Rails.env.test? ? 1 : 10
+
+  # Setup a pepper to generate the encrypted password.
+  config.pepper = '31c4a435390afd8a4dfd4a37e72d140d74219bdf2137fc09fbf36a690281c3b71fdee4bc6a09aa105848173a36be6198e3c227365e06d21535373cff25c4db26'
```
Comment (bmwwoo) [inline config/initializers/devise.rb]: same here

File: test/models/user_test.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class UserTest < ActiveSupport::TestCase
+  # test "the truth" do
+  #   assert true
+  # end
```
Comment (bmwwoo) [inline test/models/user_test.rb]: is this zombie code or is it the template for unit tests?

Comment (chriskonstad) [inline test/models/user_test.rb]: it's template for unit tests, autogenerated

File: config/initializers/devise.rb (hunk 1 above)
Comment (apfong) [inline config/initializers/devise.rb]: Good idea

File: config/initializers/devise.rb (hunk 2 above)
Comment (apfong) [inline config/initializers/devise.rb]: :+1:

File: test/models/user_test.rb (hunk 3 above)
Comment (apfong) [inline test/models/user_test.rb]: not sure what zombie code is, but thats like a template for tests

**Pull request closed without merging**